pwiz_tools/Skyline/Model/AbstractDdaSearchEngine.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using pwiz.Skyline.Model.DocSettings;
+
+namespace pwiz.Skyline.Model
+{
+    public abstract class AbstractDdaSearchEngine
+    {
+        public abstract string[] FragmentIons { get; }
+        public abstract string EngineName { get; }
+        public abstract Bitmap SearchEngineLogo { get; }
+        protected string[] SpectrumFileNames { get; set; }
+        protected string[] FastaFileNames { get; set; }
+
+        public abstract void SetMS1Tolerance(double mass, string unit);
+        public abstract void SetMS2Tolerance(double mass, string unit);
```

Comment:
Should we plan now to support MS3 searches? I think they are typically handled similarly but with different mods selected, right? So this should be MSn tolerance or product/fragment ion tolerance.

Reply:
good point, renamed the properties to precursor and fragmentIon tolerance such that each search engine can handle the tolerances in its own way!